x/genutil/module.go

```diff
@@ -106,3 +111,35 @@ func (am AppModule) ExportGenesis(_ sdk.Context, cdc codec.JSONCodec) json.RawMe
 
 // ConsensusVersion implements AppModule/ConsensusVersion.
 func (AppModule) ConsensusVersion() uint64 { return 1 }
+
+func init() {
+	appmodule.Register(&modulev1.Module{},
+		appmodule.Provide(provideModuleBasic, provideModule),
+	)
+}
+
+func provideModuleBasic() runtime.AppModuleBasicWrapper {
+	return runtime.WrapAppModuleBasic(AppModuleBasic{})
+}
+
+type genutilInputs struct {
+	depinject.In
+
+	AccountKeeper types.AccountKeeper `key:"cosmos.auth.v1.AccountKeeper"`
+	StakingKeeper types.StakingKeeper `key:"cosmos.staking.v1.Keeper"`
+	Config        client.TxConfig
+}
+
+func provideModule(in genutilInputs) (runtime.AppModuleWrapper, runtime.BaseAppOption) {
+	// this is a total hack to get access to baseapp
+	var bApp *baseapp.BaseApp
+	m := NewAppModule(in.AccountKeeper, in.StakingKeeper, func(tx abci.RequestDeliverTx) abci.ResponseDeliverTx {
+		if bApp == nil {
+			panic("BaseApp not initialized!")
+		}
+		return bApp.DeliverTx(tx)
```

Comment:
@aaronc do you think there's a less hacky way of achieving this?

Reply:
The runtime module could provide some `DeliverTx` function into the container